src/System.Data.SqlClient/tests/ExceptionTest.cs

```diff
@@ -0,0 +1,148 @@
+        private const string execReaderFailedMessage = "ExecuteReader requires an open and available Connection. The connection's current state is closed.";
+        private const string orderIdQuery = "select orderid from orders where orderid < 10250";
+
+
+        [Fact]
+        public void RunDataTest()
+        {
+            ExceptionTests(conn_string);
+            IndependentConnectionExceptionTest(conn_string);
+            VariousExceptionTests(conn_string);
+        }
+        
+
+        private void ExceptionTests(string connectionString)
+        {
+            SqlConnectionStringBuilder builder = new SqlConnectionStringBuilder(connectionString);
+
+            // tests improper server name thrown from constructor of tdsparser
+            SqlConnectionStringBuilder badBuilder = new SqlConnectionStringBuilder(builder.ConnectionString) { DataSource = badServer, ConnectTimeout = 1 };
+            
+            VerifyConnectionFailure<SqlException>(() => GenerateConnectionException(badBuilder.ConnectionString), sqlsvrBadConn, VerifyException);
+        }
+
+        private void VariousExceptionTests(string connectionString)
+        {
+            // Test exceptions - makes sure they are only thrown from upper layers
+            SqlConnectionStringBuilder builder = new SqlConnectionStringBuilder(connectionString);
+            
+            SqlConnectionStringBuilder badBuilder = new SqlConnectionStringBuilder(builder.ConnectionString) { DataSource = badServer, ConnectTimeout = 1 };
+            using (var sqlConnection = new SqlConnection(badBuilder.ConnectionString))
+            {
+                using (SqlCommand command = sqlConnection.CreateCommand())
+                {
+                    command.CommandText = orderIdQuery;
+                    VerifyConnectionFailure<InvalidOperationException>(() => command.ExecuteReader(), execReaderFailedMessage);
+                }
+            }
+        }
+
+        private void IndependentConnectionExceptionTest(string connectionString)
+        {
+            // Test exceptions for existing connection to ensure proper exception and call stack
+            SqlConnectionStringBuilder builder = new SqlConnectionStringBuilder(connectionString);
+
+            SqlConnectionStringBuilder badBuilder = new SqlConnectionStringBuilder(builder.ConnectionString) { DataSource = badServer, ConnectTimeout = 1 };
+            using (var sqlConnection = new SqlConnection(badBuilder.ConnectionString))
+            {
+                // Test 1
+                VerifyConnectionFailure<SqlException>(() => sqlConnection.Open(), sqlsvrBadConn, VerifyException);
+
+                // Test 2
```

Comment:
Is that possible to separate this tests into 2 ?

Ideally unit tests should only have 1 thing to tests.
The reason is it will be easier to pin point which test are failing rather then dig through call stack.

Some quick reference
http://www.developer.com/mgmt/top-five-best-practices-for-writing-unit-test-scripts.html